mitiq/shadows/shadows_utils.py

```diff
@@ -0,0 +1,161 @@
+
+    Returns:
+        An integer that gives the number of snapshots required to satisfy the
+        shadow bound.
+    """
+    return int(34 * (4**num_qubits) * epsilon ** (-2))
+
+
+def local_clifford_shadow_norm(opt: PauliString[Any]) -> float:
+    """
+    Calculate shadow norm of an operator with random unitary sampled from local
+    Clifford group.
+    Args:
+        opt: a self-adjoint operator
+    Returns:
+        Shadow norm when unitary ensemble is local Clifford group.
+    """
+    norm = (
+        np.linalg.norm(
+            cirq.unitary(opt)
+            - np.trace(cirq.unitary(opt))
+            / 2 ** int(np.log2(cirq.unitary(opt).shape[0])),
+            ord=np.inf,
+        )
+        ** 2
+    )
+    return float(norm)
+
+
+def n_measurements_opts_expectation_bound(
+    error: float,
+    observables: List[PauliString[Any]],
+    failure_rate: float,
+) -> Tuple[int, int]:
+    """
+    This function returns the minimum number of classical shadows required and
+    the number of groups "k" into which we need to split the shadows for
+    achieving the desired accuracy and failure rate in operator expectation
+    value estimation.
+
+    Args:
+        error: The error on the estimator.
+        observables: List of cirq.PauliString corresponding to the
+            observables we intend to measure.
+        failure_rate: Rate of failure for the bound to hold.
+
+    Returns:
+        Integers quantifying the number of snapshots required to satisfy
+        the shadow bound and the chunk size required to attain the specified
+        failure rate.
+    """
+    M = len(observables)
+    K = 2 * np.log(2 * M / failure_rate)
+
+    N = (
+        34
+        * max(local_clifford_shadow_norm(o) for o in observables)
+        / error**2
+    )
+    return int(np.ceil(N * K)), int(K)
+
+
+def fidelity(
+    state_vector: NDArray[np.complex64],
+    rho: NDArray[np.complex64],
+) -> float:
+    """
+    Calculate the fidelity between a state vector and a density matrix.
+    Args:
+        state_vector: The vector whose norm we want to calculate.
+        rho: The operator whose norm we want to calculate.
+
+    Returns:
+        Scalar corresponding to the fidelity.
+    """
+    return np.reshape(state_vector.conj().T @ rho @ state_vector, -1).real[0]
+
+
+def transform_to_cirq_paulistring(
+    pauli_str: Union[str, mitiq.PauliString, cirq.PauliString[Any]]
+) -> Tuple[float, cirq.PauliString[Any]]:
+    """Transforms mitiq.PauliString, cirq.PauliString, cirq.PauliSum or string
+    to a cirq.PauliString class.
+
+    Args:
+      pauli_str: A mitiq.PauliString, cirq.PauliString, cirq.PauliSum
+        or string.
+
+    Returns:
+      A tuple containing the coefficient and a cirq.PauliString.
+    """
+    if isinstance(pauli_str, cirq.PauliString):
+        # Extract coefficient and return cirq.PauliString without coefficient
+        coeff = np.real(pauli_str.coefficient)  # type: ignore
```

Comment:
Coefficients can be complex and the user is not aware that we'll take the real part.
Is it necessary to take the real part? Can't we keep a complex coefficient instead?

With the current code we get:
`Paulistring(cirq.X, coefficient=1.0j) --(this function)-->Paulistring(cirq.X, coefficient=0.0) --> 0`

Reply:
I was taught that the so-called observable should be a self-adjoint operator. But I have changed this to allow complex coeff.

Reply:
Sometimes it can be useful to consider operators with complex coefficients even if they are not  "observables" in the physical sense. This is why a `cirq.PauliString` (or `mitiq.PauliString`) object  can be defined with a complex coefficient. 

Personally, I also think this aspect is a bit confusing but unless we revisit how the classes `cirq.PauliString` or `mitiq.PauliString` are defined, we need to assume the possibility of complex coefficients.

Reply:
Ahh, I see, thanks a lot for your explanation, Andrea!